hooks/google_analytics_hook.py

```diff
@@ -20,37 +20,81 @@
 "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
 "client_x509_cert_url": "{CERT_URL}"
 
+In Airflow 1.9.0 this requires to use the web interface or cli to set connection extra's. If you prefer to not use the
+web interface to manage connections you can also supply the key as a json file.
+
+@TODO: add support for p12 keys
+
 More details can be found here:
 https://developers.google.com/api-client-library/python/guide/aaa_client_secrets
 """
 
-from airflow.hooks.base_hook import BaseHook
+import time
+import os
 
+from airflow.hooks.base_hook import BaseHook
+from airflow import configuration as conf
 from apiclient.discovery import build
+from apiclient.http import MediaInMemoryUpload
 from oauth2client.service_account import ServiceAccountCredentials
 from oauth2client.client import AccessTokenCredentials
-import time
+from collections import namedtuple
 
 
 class GoogleAnalyticsHook(BaseHook):
-    def __init__(self, google_analytics_conn_id='google_analytics_default'):
+    GAService = namedtuple('GAService', ['name', 'version', 'scopes'])
+    # We need to rely on 2 services depending on the task at hand: reading from or writing to GA.
+    _services = {
+        'reporting': GAService(name='analyticsreporting',
+                               version='v4',
+                               scopes=['https://www.googleapis.com/auth/analytics.readonly']),
+        'management': GAService(name='analytics',
+                                version='v3',
+                                scopes=['https://www.googleapis.com/auth/analytics'])
+    }
+    _key_folder = os.path.join(conf.get('core', 'airflow_home'), 'keys')
+
+    def __init__(self, google_analytics_conn_id='google_analytics_default', key_file=None):
         self.google_analytics_conn_id = google_analytics_conn_id
         self.connection = self.get_connection(google_analytics_conn_id)
+        if 'client_secrets' in self.connection.extra_dejson:
+            self.client_secrets = self.connection.extra_dejson['client_secrets']
+        if key_file:
+            self.file_location = os.path.join(GoogleAnalyticsHook._key_folder, key_file)
 
-        self.client_secrets = self.connection.extra_dejson['client_secrets']
+    def get_service_object(self, name):
```

Comment:
I might default to either `reporting` or `management` or handle the error if a third value gets passed in.